src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+
+        return 1 + (doy % InternationalFixedChronology.DAYS_IN_MONTH);
+    }
+
+    private int getCalculatedDayOfMonth() {
+        return isYearDay() ? InternationalFixedChronology.DAYS_IN_MONTH + 1 : isLeapDay() ? 1 : getDayOfMonth();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getDayOfYear() {
+        return dayOfYear;
+    }
+
+    private int getDayOfYearAdjusted() {
+        return isLeapYear() && dayOfYear > 6 * InternationalFixedChronology.DAYS_IN_MONTH ? dayOfYear - 1 : dayOfYear;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int lengthOfYearInMonths() {
+        return InternationalFixedChronology.MONTHS_IN_YEAR;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    ValueRange rangeAlignedWeekOfMonth() {
+        return ValueRange.of(1, InternationalFixedChronology.WEEKS_IN_MONTH);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    InternationalFixedDate resolvePrevious(final int newYear, final int newMonth, final int dayOfMonth) {
+        return resolvePreviousValid(newYear, newMonth, dayOfMonth);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Gets the chronology of this date, which is the International fixed calendar system.
+     * <p/>
+     * The {@code Chronology} represents the calendar system in use.
+     * The era and other fields in {@link ChronoField} are defined by the chronology.
+     *
+     * @return the International fixed chronology, not null
+     */
+    @Override
+    public InternationalFixedChronology getChronology() {
+        return InternationalFixedChronology.INSTANCE;
+    }
+
+    /**
+     * Gets the era applicable at this date.
+     * <p/>
+     * The International fixed calendar system only has one era, 'CE',
+     * defined by {@link InternationalFixedEra}.
+     *
+     * @return the era applicable at this date, not null
+     */
+    @Override
+    public InternationalFixedEra getEra() {
+        return InternationalFixedEra.CE;
+    }
+
+    /**
+     * Returns the length of the month represented by this date.
+     * <p/>
+     * This returns the length of the month in days.
+     * Month lengths do not match those of the ISO calendar system.
+     *
+     * @return the length of the month in days: 28
+     */
+    @Override
+    public int lengthOfMonth() {
+        return InternationalFixedChronology.DAYS_IN_MONTH;
+    }
+
+    /**
+     * Returns the length of the year represented by this date.
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    @Override
+    public InternationalFixedDate plus(final long amountToAdd, final TemporalUnit unit) {
```

Comment:
Need to take special care with weeks, given that leap-day and year-day aren't part of any week.  See Discordian.